src/sentry/replays/usecases/ingest/event_parser.py

```diff
@@ -436,79 +454,115 @@ def as_trace_item_context(event_type: EventType, event: dict[str, Any]) -> Trace
-                    "value": float(payload["data"]["value"]),
-                },
+                "attributes": web_vital_attributes,
                 "event_hash": uuid.uuid4().bytes,
-                "timestamp": float(payload["startTimestamp"]),
+                "timestamp": float(payload.get("startTimestamp", 0)),
             }
+
         case EventType.HYDRATION_ERROR:
-            payload = event["data"]["payload"]
+            payload = event.get("data", {}).get("payload", {})
+            payload_data = payload.get("data", {})
+
+            hydration_attributes = {
+                "category": "replay.hydrate-error",
+            }
+            if "url" in payload_data:
+                hydration_attributes["url"] = as_string_strict(payload_data["url"])
+
             return {
-                "attributes": {
-                    "category": "replay.hydrate-error",
-                    "url": as_string_strict(payload["data"]["url"]),
-                },
+                "attributes": hydration_attributes,
                 "event_hash": uuid.uuid4().bytes,
-                "timestamp": float(event["data"]["payload"]["timestamp"]),
+                "timestamp": float(payload.get("timestamp", 0)),
             }
+
         case EventType.MUTATIONS:
-            payload = event["data"]["payload"]
+            payload_data = event.get("data", {}).get("payload", {}).get("data", {})
+
+            mutations_attributes = {
+                "category": "replay.mutations",
+            }
+            if "count" in payload_data:
+                mutations_attributes["count"] = int(payload_data["count"])
+
             return {
-                "attributes": {
-                    "category": "replay.mutations",
-                    "count": int(payload["data"]["count"]),
-                },
+                "attributes": mutations_attributes,
                 "event_hash": uuid.uuid4().bytes,
-                "timestamp": event["timestamp"],
+                "timestamp": float(event.get("timestamp", 0)),
             }
         case EventType.UNKNOWN:
             return None
         case EventType.CANVAS:
             return None
         case EventType.OPTIONS:
-            payload = event["data"]["payload"]
+            payload = event.get("data", {}).get("payload", {})
+
+            options_attributes = {
+                "category": "sdk.options",
+            }
+            for key in [
+                "shouldRecordCanvas",
+                "useCompressionOption",
+                "blockAllMedia",
+                "maskAllText",
+                "maskAllInputs",
+                "useCompression",
+                "networkDetailHasUrls",
+                "networkCaptureBodies",
+                "networkRequestHasHeaders",
+                "networkResponseHasHeaders",
+            ]:
+                if key in payload:
+                    options_attributes[key] = bool(payload[key])
```

Comment:
One thing we lose from this is presence validation.  We're relaxing our output schema and are saying "give us 0 or more of these items".  Whereas before we were maximally demanding "give us everything or this fails".  Ideally we're as maximally demanding as is pragmatic.  That allows us to screen malformed or incomplete events while also cluing us in to input schema changes.

With this change we'll never err again but we'll also never be able to correct our parser because we'll have no idea its malfunctioning.